osu.Game/Screens/Select/FilterQueryParser.cs

```diff
@@ -312,11 +323,45 @@ private static bool tryUpdateCriteriaRange<T>(ref FilterCriteria.OptionalRange<T
 
         private static bool tryUpdateLengthRange(FilterCriteria criteria, Operator op, string val)
         {
-            if (!tryParseDoubleWithPoint(val.TrimEnd('m', 's', 'h'), out double length))
+            List<string> parts = new List<string>();
+            GroupCollection groups = null;
+
+            if (
+                tryMatchRegex(val, @"^((?<hours>\d+):)?(?<minutes>\d+):(?<seconds>\d+)$", ref groups) ||
+                tryMatchRegex(val, @"^((?<hours>\d+(\.\d+)?)h)?((?<minutes>\d+(\.\d+)?)m)?((?<seconds>\d+(\.\d+)?)s)?$", ref groups) ||
+                tryMatchRegex(val, @"^(?<seconds>\d+(\.\d+)?)$", ref groups)
```

Comment:
This all can't be in an `if` condition, it just reads wrong. Recommended pattern to use instead:

```csharp
Match? match = null;

match ??= tryMatchRegex(val, ...);
match ??= tryMatchRegex(val, ...);

if (match != null)
    // continue processing
```